app/src/main/java/org/simple/clinic/sync/DataSync.kt

```diff
@@ -9,12 +12,25 @@ import timber.log.Timber
 import javax.inject.Inject
 
 class DataSync @Inject constructor(
-    private val syncs: ArrayList<ModelSync>,
+    private val modelSyncs: ArrayList<ModelSync>,
     private val crashReporter: CrashReporter
 ) {
 
   fun sync(): Completable {
-    val completables = syncs.map { it.sync() }
+    return runAndDelayError(modelSyncs.map { it.sync() })
+  }
+
+  fun syncGroup(syncGroupId: String): Completable {
+    return Observable
+        .fromIterable(modelSyncs)
+        .flatMapSingle { modelSync -> modelSync.syncConfig().zipWith(Single.just(modelSync)) }
+        .filter { (config, _) -> config.syncGroupId == syncGroupId }
+        .map { (_, modelSync) -> modelSync.sync() }
+        .toList()
+        .flatMapCompletable(this::runAndDelayError)
+  }
+
+  private fun runAndDelayError(completables: List<Completable>): Completable {
```

Comment:
`runAndDelayError ()` -> `runAndSwallowError()`? Because this function doesn't just delay error.

Reply:
Done.